core/sql/src/main/java/org/trafodion/sql/HTableClient.java

```diff
@@ -358,6 +365,417 @@ void resetAutoFlush() {
+									columnsToRemove,
+									true), //filterIfMissing,
+							SingleColumnValueExcludeOrNotFilter(
+									(byte[])colNamesToFilter[k], 
+									CompareOp.GREATER_OR_EQUAL, 
+									new BinaryComparator((byte[])colValuesToCompare[kk]),
+									columnsToRemove)));
+					k++;kk++;
+					break;
+				case NO_OP:
+					linkedList.addLast(SingleColumnValueExcludeOrNotFilter(
+							(byte[])colNamesToFilter[k], 
+							CompareOp.NO_OP, 
+							new BinaryComparator((byte[])colValuesToCompare[kk]),
+							columnsToRemove));
+					k++;kk++;
+					break;
+				case NO_OP_NULL:
+					linkedList.addLast(new FilterList(FilterList.Operator.MUST_PASS_ALL, //AND between if not null and the actual
+							SingleColumnValueExcludeOrNotFilter(
+									(byte[])colNamesToFilter[k],
+									CompareOp.EQUAL, 
+									new BinaryPrefixComparator(new byte[]{0x00}),//check for null indicator = 0 representing non null
+									columnsToRemove,
+									true), //filterIfMissing,
+							SingleColumnValueExcludeOrNotFilter(
+									(byte[])colNamesToFilter[k], 
+									CompareOp.NO_OP, 
+									new BinaryComparator((byte[])colValuesToCompare[kk]),
+									columnsToRemove)));
+					k++;kk++;					
+					break;
+				case IS_NULL:
+					// is null on a non nullable column!
+					linkedList.addLast(SingleColumnValueExcludeOrNotFilter(
+							(byte[])colNamesToFilter[k], 
+							CompareOp.NO_OP, //exclude everything
+							new BinaryPrefixComparator((new byte[]{})),
+							columnsToRemove));
+					k++;
+					break;
+				case IS_NULL_NULL:
+					// is_null on nullable column: is absent OR has the first byte set to FF indicating NULL.
+					linkedList.addLast(
+							new FilterList(FilterList.Operator.MUST_PASS_ONE, //OR
+									SingleColumnValueExcludeOrNotFilter(
+											(byte[])colNamesToFilter[k],
+											CompareOp.EQUAL, 
+											new NullComparator(),//is absent?
+											columnsToRemove), 
+									SingleColumnValueExcludeOrNotFilter(
+											(byte[])colNamesToFilter[k],
+											CompareOp.EQUAL, 
+											new BinaryPrefixComparator(new byte[]{-1}),//0xFF has null prefix indicator
+											columnsToRemove)));
+					k++;
+					break;
+				case IS_NOT_NULL:
+					// is not null on a non nullable column!
+					// do nothing, always true
+					k++;
+					break;	
+				case IS_NOT_NULL_NULL:
+					// is_not_null on nullable column: is not absent AND has the first byte not set to FF indicating NULL.
+					linkedList.addLast(SingleColumnValueExcludeOrNotFilter(
+							(byte[])colNamesToFilter[k],
+							CompareOp.NOT_EQUAL, 
+							new BinaryPrefixComparator(new byte[]{-1}),// 0xFF has null prefix indicator
+							columnsToRemove,
+							true));//filter if missing (if absent null)
+					k++;
+					break;
+				case AND:
+					linkedList.addLast("AND");
+					break;
+				case OR:
+					linkedList.addLast("OR");
+					break;
+					default:
+			}//switch
+		}//for
```

Comment:
This for loop just puts items in reverse order.  Where do we do the in-order traversal?